### PR TITLE
:lock: Add authorization to manage page

### LINKED ISSUE #97: Add authorization for /manage page
## Summary

Manage page should only be visible to admins. Add authorization to secure that page

## Motivation

Manage page has sensitive information that's not for every user to see

## Additional context

Right now there's only two roles: user and admin, maybe could add a third one that's has only read access

### PR DESCRIPTION
**Pull request by adlaiCarmona**
### Description of the Change

Only Admin users can enter to manage page

### Verification Process

With admin account
1. Run the UI
2. Must be able to see manage on the navbar
3. Go to manage page and should be able to see data

With user account
1. Run the UI
2. Shouldn't be able to see manage on the navbar
3. Go to manage page and should be redirected to index

### Applicable Issues

- Closes #97 

### Release Notes

Add Authorization to manage and roles

### CODE REVIEW

File: prisma-erd.png
Comment (JordiEspinozaMendoza) [inline prisma-erd.png:1]: Do we need this image in the repo? Or can we add this file to the gitgnore file

File: prisma/schema.prisma [hunk 1]
```diff
@@ -48,6 +48,7 @@ model User {
   name          String
   email         String        @unique
   emailVerified DateTime?
+  role          Int        @default(0)
```
Comment (JordiEspinozaMendoza) [inline prisma/schema.prisma:51]: What is the role 0?

File: src/server/auth.ts [hunk 2]
```diff
@@ -19,15 +19,17 @@ declare module "next-auth" {
...
   }
 
-  // interface User {
-  //   // ...other properties
-  //   // role: UserRole;
-  // }
+  interface User {
+    role: number;
+    // ...other properties
+    // role: UserRole;
```
Comment (JordiEspinozaMendoza) [inline src/server/auth.ts:31]: Do we need this comments?

File: prisma-erd.png
Comment (adlaiCarmona) [reply prisma-erd.png:1]: The image is for record tracking. We are not really beign audited, but if we need to check the status of the schemas in a given point, it's easier to check the image to see the relations between tables.

File: prisma/schema.prisma (hunk 1 above)
Comment (adlaiCarmona) [reply prisma/schema.prisma:51]: a normal user would be a 0, an admin would be a 1, maybe we could add a reader-admin role, but could add that easy with the enum.ts

File: src/server/auth.ts (hunk 2 above)
Comment (adlaiCarmona) [reply src/server/auth.ts]: Comments removed

**Review by JordiEspinozaMendoza — APPROVED** [review]

**Review by KatherynneCPlessmann — APPROVED** [review]

**Pull request merged**